content/Rails - Multiple Databases.md

```diff
@@ -0,0 +1,124 @@
+```ruby
+ActiveRecord::Base.connected_to(role: :reading) do
+  # all code in this block will be connected to the reading role
+end
+```
+
+- Automatic swapping between the primary and replica depending on the HTTP verb and recent writes
+
+As we are used to in Rails, this comes standardized:
+
+“If the application is receiving a POST, PUT, DELETE, or PATCH request the application will automatically write to the primary. For the specified time after the write, the application will read from the primary. For a GET or HEAD request, the application will read from the replica unless there was a recent write.“
+
+- Rails tasks for creating, dropping, migrating, and interacting with the multiple databases
+
+First big comment, the usual `rails db:create` will now create both databases, as so the usual `rails db:migrate` will migrate both databases too.
+You need to specify which database you want to work with like `rails db:create:primary` or `rails db:migrate:secondary` if you want to just modify one.
+
+So for your custom tasks you may need to also specify the desired database you are working with.
+
+#### Pending ones
+
+- Sharding
+Right now there are some gems to accomplish this, so until is integrated onto Rails we should keep using gems (no real deal with it)
+
+- Joining across clusters
+This is what Rails state:
+“Applications cannot join across databases. Rails 6.1 will support using has_many relationships and creating 2 queries instead of joining, but Rails 6.0 will require you to split the joins into 2 selects manually.“
+
+So we just need to wait for rails 6.1!
+
+- Load balancing replicas
+This seems an interesting feature, so I hope to see upcoming news for this.
+
+- Dumping schema caches for multiple databases
+Right now this is what is stated:
+“If you use a schema cache and multiple databases you'll need to write an initializer that loads the schema cache from your app. This wasn't an issue we could resolve in time for Rails 6.0 but hope to have it in a future version soon“
+
+So seems like a trivial issue to solve in upcoming versions
+
+
+## Rails 6.1 upcoming state
+
+As you may know, rails 6.1 is almost there so I will double-check what was stated to be released for this new version, and what is actually going to be released
+
+Nice open PRs right now:
+- [Infer migrations_paths from database name](https://github.com/rails/rails/pull/36886)
+- [Sharding + configuration DSL](https://github.com/rails/rails/pull/38721)
+
+So the summary would be:
+- Sharding is now a WIP thing and we may hope to see it live in Rails 6.1
+- Still waiting for the Joining across clusters (or I missed it in my review)
+- Still waiting for load balancing
+- Schema caches are now a thing!
+
+## Some personal ideas
+
+I'm a huge fan of non-relational databases, so I would love to see some mixins of MongoDB + Postgres for example. And with this introduced changes I think that the road is prepared enough to start working on a way to properly work with it.
```

Comment:
I would say `with these changes introduced` (`these` instead of `this` and `introduced` after changes).